invenio_users_resources/services/generators.py

```diff
@@ -108,3 +111,31 @@ def query_filter(self, identity=None, **kwargs):
                     return dsl.Q("term", id=need.value)
 
         return []
+
+
+class AdminAction(Generator):
```

Comment:
Comment: The ``AdminAction`` should likely be moved higher up the hierarchy - invenio-records-permissions or Invenio-access.

Reply:
I moved it up to `invenio-records-permissions` and renamed it to `AllowedByAction` since it's an action-based generator (Admin seemed a bit out of context).